File: Python/bubble_sort.py
# Binary Sorting algorithm written in Python.

def bubble_sort(items):
    temp = None
    for i in range(1, len(items)):
        for j in range(0, len(items) - i):
            if items[j] > items[j + 1]:
                temp = items[j]
                items[j] = items[j + 1]
                items[j + 1] = temp
    return items

File: C++/bubble_sort.cpp
#include <iostream>
#include <vector>
using namespace std;

int main() {

    vector<int> data { 5, 1, 3, 2, 4 };
    int temp {};
    for (int i = 1; i < data.size(); i++) {
        for (int j = 0; j < data.size() - i; j++) {
            if (data[j] > data[j + 1]) {
                temp = data[j];
                data[j] = data[j + 1];
                data[j + 1] = temp;
            }
        }
    }
    return 0;
}